### DISCUSSION TITLE
Adding to Hydra's searchpath in the compose API?

### DISCUSSION THREAD

**Discussion by stweigand97** [facebookresearch/hydra#2834] [Q&A] [answered]
Is it possible to add multiple modules or directories to Hydra's searchpath when using the compose API?

I am looking for a way to dynamically modify Hydra's searchpath, similar to the ``--config-dir`` option, just not with the CLI.

**Comment by odelalleau** [comment]
Have you looked into writing a [Search Path plugin](https://github.com/facebookresearch/hydra/blob/9c522a073b926704580afbd86c7b75613d6c9008/examples/plugins/example_searchpath_plugin/hydra_plugins/example_searchpath_plugin/example_searchpath_plugin.py#L6)?

**Reply by stweigand97** [reply]
Yes, but I don't think these are suited in this case. I want to add a searchpath that is decided at runtime. The paths added through a search path plugin must be fixed in the class definition due to the way that hydra loads these, no?

**Reply by odelalleau** [reply, marked as answer]
I think it should be possible to [register a plugin](https://github.com/facebookresearch/hydra/blob/30511140960aa43e6d649581f13d30f03254f50f/hydra/core/plugins.py#L77) before calling the compose API.
